app/controllers/mixins/middleware_common_mixin.rb

```diff
@@ -25,4 +33,26 @@ def clear_topology_breadcrumb
       @breadcrumbs.clear
     end
   end
+
+  def show_middleware(record, controller_name, display_name)
+    return if record_no_longer_exists?(record)
+    @gtl_url = "/show"
+    drop_breadcrumb({:name => display_name,
+                     :url  => "/#{controller_name}/show_list?page=#{@current_page}&refresh=y"},
+                    true)
+    breadcrumb_url = "/#{controller_name}/show/#{record.id}?refresh=n&display=#{@display}"
+    case @display
+    when 'download_pdf', 'main', 'summary_only' then show_summary(@display, controller_name, @record)
+    when 'timeline' then show_timeline(@record, breadcrumb_url)
+    when 'performance' then show_performance(breadcrumb_url)
+    end
+  end
+
+  def show_middleware_display(record, display, clazz, alt_controller_name = nil)
```

Comment:
what is the difference between this method and `show_middleware`? when this one should be used versus the other one?